src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/transaction_guarantee.rs

```diff
@@ -2,8 +2,81 @@ use crate::prelude::*;
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash, uniffi::Record)]
 pub struct TransactionGuarantee {
-    pub amount: Decimal192,
+    amount: Decimal192,
     pub instruction_index: u64,
     pub resource_address: ResourceAddress,
-    pub resource_divisibility: Option<i64>,
+    pub resource_divisibility: Option<i32>,
```

Comment:
Why do we need this to be optional?

Reply:
it need not be optional, but letting it be says "use default" and it is how it was impl on iOS.